lib/ai_bot/tools/tool.rb

```diff
@@ -77,6 +77,35 @@ def standalone?
 
         protected
 
+        def send_http_request(url, headers: {}, authenticate_github: false)
```

Comment:
Since this is generic, we shouldn't specifically reference github. The `request["Authorization"] = "Bearer #{SiteSetting.ai_bot_github_access_token}"` should just be passed in as part of `headers`

Reply:
@SamSaffron will you change this in a separate PR?